apps/agentic-chat/src/hooks/useTools.ts

```diff
@@ -93,7 +97,23 @@ const useTools = () => {
         buyAssetId,
         sellAmountCryptoPrecision,
         fromAddress: getAddress(account?.address ?? ''),
-        setBebopQuote,
+        setQuote,
+        assetsStore,
+      });
+    },
+  });
+
+  useAssistantTool({
+    toolName: 'relayRate',
+    description: 'Fetches a swap rate from Relay and displays it to the user',
+    parameters: relayRateParams,
+    execute: async ({ sellAssetId, buyAssetId, sellAmountCryptoPrecision }) => {
+      return getRelayRate({
+        sellAssetId,
+        buyAssetId,
+        sellAmountCryptoPrecision,
+        fromAddress: getAddress(account?.address ?? ''),
+        setQuote,
```

Comment:
Should we already split rates and quotes?

What happens if the user ask to execute it but 10 or 20 minutes later?